src/app/util/ember-compatibility-functions.cpp

```diff
@@ -421,6 +454,19 @@ CHIP_ERROR prepareWriteData(EmberAfAttributeType expectedType, TLV::TLVReader &
         return numericTlvDataToAttributeBuffer<int8_t>(aReader, dataLen);
     case ZCL_INT16S_ATTRIBUTE_TYPE: // Signed 16-bit integer
         return numericTlvDataToAttributeBuffer<int16_t>(aReader, dataLen);
+    case ZCL_INT24S_ATTRIBUTE_TYPE: // Signed 24-bit integer
+    {
+        int32_t value;
+        static_assert(3 <= sizeof(attributeData), "Value cannot fit into attribute data");
+        ReturnErrorOnFailure(aReader.Get(value));
+        if (value > static_cast<int32_t>((1UL << ((3 * 8) - 1)) - 1) || value < static_cast<int32_t>(-(1UL << ((3 * 8) - 1))))
```

Comment:
```suggestion
        if (value > static_cast<int32_t>((1UL << ((3 * 8) - 1)) - 1) || value < static_cast<int32_t>(-(1UL << ((3 * 8) - 1) ) + 1))
```
The most-negative value representable in 24 bits is not an allowed value, right?

Reply:
The 2. complement values of a 24-bit signed value are:
-2^(24-1) to +(2^(24-1)-1)
Which is -8388608 to 8388607
Just like the 8-bit version is:
-2^(8-1) to +(2^(8-1)-1) ==> -128 to 127
At least to the best of my knowlegde.

Reply:
The point is, `-2^(24-1)` is not a valid uint24 value in the spec.